File: src/vibe_kanban_clone/schemas/skill.py
"""Skill schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class SkillCreate(BaseModel):
    """Schema for creating a skill."""

    name: str = Field(min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)
    instructions: str | None = Field(default=None, min_length=1, max_length=10000)
    allowed_tools: list[Any] | None = None


class SkillRead(BaseModel):
    """Schema for reading a skill."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None
    instructions: str | None
    allowed_tools: list[Any] | None
    created_at: datetime


class SkillUpdate(BaseModel):
    """Schema for updating a skill."""

    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)
    instructions: str | None = Field(default=None, min_length=1, max_length=10000)
    allowed_tools: list[Any] | None = None

File: server/src/schemas/skill.ts
import { z } from "zod";

export const SkillCreate = z.object({
  name: z.string().min(1).max(255),
  description: z.string().optional(),
  instructions: z.string().optional(),
  allowed_tools: z.array(z.string()).optional(),
});

export const SkillRead = z.object({
  id: z.number(),
  name: z.string(),
  description: z.string().nullable(),
  instructions: z.string().nullable(),
  allowed_tools: z.array(z.string()).nullable().optional(),
  created_at: z.string().or(z.date()),
});

export const SkillUpdate = z.object({
  name: z.string().min(1).max(255).optional(),
  description: z.string().optional(),
  instructions: z.string().optional(),
  allowed_tools: z.array(z.string()).optional(),
});

export type SkillCreate = z.infer<typeof SkillCreate>;
export type SkillRead = z.infer<typeof SkillRead>;
export type SkillUpdate = z.infer<typeof SkillUpdate>;
